### PR TITLE
UT for global_function in module_base

### LINKED ISSUE #777: I will add the UTs for global_functions in module_base
## Background
ABACUS lack UT for global_functions in module_base.

## Describe the solution you'd like
We should add UT for global_functions in module_base

### PR DESCRIPTION
**Pull request by hongriTianqi**

### CODE REVIEW

File: source/module_base/global_function.cpp [hunk 1]
```diff
@@ -2,52 +2,55 @@
...
+    {
+        // GlobalV::ofs_running << " *********************************************************************************"
+        // << std::endl;
+        GlobalV::ofs_running << " !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
+                             << std::endl;
+        GlobalV::ofs_running << " " << words << std::endl;
+        GlobalV::ofs_running << " !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
+                             << std::endl;
+    }
+    return;
```
Comment (dyzheng) [inline source/module_base/global_function.cpp]: Change the place of "return;" would change output files. 
It seems like this function is falling into disuse.
Maybe keeping this function for doing nothing would be better?

File: source/module_base/global_function.cpp [hunk 2]
```diff
@@ -112,7 +115,8 @@ void DONE(std::ofstream &ofs,const std::string &description, const bool only_ran
 void TEST_LEVEL(const std::string &name)
 {
     bool disable = true;
-    if (disable) return;
+    if (disable)
+        return;
```
Comment (dyzheng) [inline source/module_base/global_function.cpp]: This format change seems unnecessary, only two formats is allowed for "if" sentence
"if(disable) return;"
or
"
if(disable)
{
    return;
}
"

File: source/module_base/global_function.cpp [hunk 3]
```diff
@@ -162,97 +164,100 @@ bool SCAN_BEGIN(std::ifstream &ifs, const std::string &TargetName, const bool re
...
-	throw std::runtime_error("read /proc/meminfo error in "+TO_STRING(__FILE__)+" line "+TO_STRING(__LINE__));
+    size_t mem_sum = 0;
+    int i = 0;
+    std::ifstream ifs("/proc/meminfo");
+    while (ifs.good())
+    {
+        std::string label, size, kB;
+        ifs >> label >> size >> kB;
+        if (label == "MemAvailable:")
+            return std::stol(size);
```
Comment (dyzheng) [inline source/module_base/global_function.cpp:249]: Please add "{}" for this line.

File: source/module_base/global_function.cpp [hunk 4]
```diff
@@ -162,97 +164,100 @@ bool SCAN_BEGIN(std::ifstream &ifs, const std::string &TargetName, const bool re
...
+        else if (label == "MemFree:" || label == "Buffers:" || label == "Cached:")
+        {
+            mem_sum += std::stol(size);
+            ++i;
+        }
+        if (i == 3)
+            return mem_sum;
```
Comment (dyzheng) [inline source/module_base/global_function.cpp]: Please add "{}" for this line.

File: source/module_base/global_function.cpp (hunk 2 above)
Comment (caic99) [reply source/module_base/global_function.cpp]: @dyzheng I think the format in proposed changing is acceptable.

**Review by caic99 — APPROVED** [review]

File: source/module_base/global_function.cpp (hunk 1 above)
Comment (caic99) [reply source/module_base/global_function.cpp]: @dyzheng I suppose we can remove those "dead code".

**Review by dyzheng — APPROVED** [review]

**Pull request merged**